### PR TITLE
eZ Launchpad documentation

### PR DESCRIPTION
**Pull request by Plopix**
Explanations on how to use eZ Launchpad with Platform.sh for eZ Platform.

Potential reviewers on eZ part: @lserwatka, @andrerom, @SylvainGuittard

PR env: https://pr-1045-rlvkyga-t2llqeifuhpzg.eu.platform.sh/frameworks/ez.html

### CODE REVIEW

File: src/frameworks/ez.md [hunk 1]
```diff
@@ -1,21 +1,146 @@
 # eZ Platform
 
-eZ Platform is a Composer-based PHP CMS, and as such fits well with the Platform.sh model.  As a Symfony-based application its setup is very similar to Symfony.
+eZ Platform is a Composer-based PHP CMS, and as such fits well with the Platform.sh model.  As a Symfony-based 
+application its setup is very similar to Symfony.
 
-eZ Platform comes pre-configured for use with Platform.sh in version 1.13 and later and version 2.0.1 and later.  Those are the only versions that are supported.  Appropriate Platform.sh configuration files are included in the eZ Platform application itself, but of course may be modified to suit your particular site if needed.
+eZ Platform comes pre-configured for use with Platform.sh in version 1.13 and later. Version is 2.x and later is
```
Comment (SylvainGuittard) [inline src/frameworks/ez.md]: eZ Platform v2.5 will be the LTS for v2.x and will be released in 2 weeks. I think we can change it for
```suggestion
eZ Platform comes pre-configured for use with Platform.sh in version 1.13 and later. Version is 2.5 and later is
```

File: src/frameworks/ez.md [hunk 2]
```diff
@@ -1,21 +1,146 @@
...
+recommended.
+Those are the only versions that are supported.  Appropriate Platform.sh configuration files are included in the
```
Comment (Crell) [inline src/frameworks/ez.md]: Is this supposed to be a new paragraph?  If so, it needs a blank line.  If not, remove the carriage return.

File: src/frameworks/ez.md [hunk 3]
```diff
@@ -1,21 +1,146 @@
...
+eZ Platform application itself, but of course may be modified to suit your particular site if needed.
 
 ## Cache and sessions
 
-By default, eZ Platform is configured to use a single Redis instance for both the application cache and session storage.  You may optionally choose to use a separate Redis instance for session storage in case you have a lot of authenticated traffic (and thus there would be many session records).
+By default, eZ Platform is configured to use a single Redis instance for both the application cache and session
+storage.  You may optionally choose to use a separate Redis instance for session storage in case you have a lot of
+authenticated traffic (and thus there would be many session records).
```
Comment (Crell) [inline src/frameworks/ez.md]: Erp, no, coding standards for docs is to not have hard wraps mid-paragraph.  Please revert this change throughout the document.

File: src/frameworks/ez.md [hunk 4]
```diff
@@ -1,21 +1,146 @@
...
  * The [.platform](https://github.com/ezsystems/ezplatform/tree/master/.platform) directory
- * The [platformsh.php](https://github.com/ezsystems/ezplatform/blob/master/app/config/env/platformsh.php) configuration file, which does the work of mapping Platform.sh environment variables into eZ Platform.  It also will automatically enable Redis-based cache and session support if detected.
+ * The [platformsh.php](https://github.com/ezsystems/ezplatform/blob/master/app/config/env/platformsh.php) configuration
+  file, which does the work of mapping Platform.sh environment variables into eZ Platform.  It also will automatically 
+  enable Redis-based cache and session support if detected.
+
+
+## Local Development with eZ Platform 2.x and later
+
+eZ systems provide a tool called [eZ Launchpad](https://ezsystems.github.io/launchpad/) for local development that
```
Comment (Crell) [inline src/frameworks/ez.md]: Shouldn't "eZ Systems" be capital S?

File: src/frameworks/ez.md [hunk 5]
```diff
@@ -1,21 +1,146 @@
...
+improves Developer eXperience and reduces the container management. eZ Launchpad is ready to work with Platform.sh.
+
+### eZ Launchpad installation
+
+eZ Launchpad approach is to stay as decoupled as possible from your development machine and your remote hosting whether
```
Comment (Crell) [inline src/frameworks/ez.md]: "ez Launchpad's", possessive.

File: src/frameworks/ez.md [hunk 6]
```diff
@@ -1,21 +1,146 @@
...
+you are Linux or Mac OSX. Installation is really simple:
```
Comment (Crell) [inline src/frameworks/ez.md]: Avoid terms like "simple" or "easy" where possible.  Here it's better to just say "To install run:"

File: src/frameworks/ez.md [hunk 7]
```diff
@@ -1,21 +1,146 @@
...
+
+```bash
+curl -LSs https://ezsystems.github.io/launchpad/install_curl.bash | bash
+```
+ 
+Then you can just start to use it to initialize your eZ Platform project on top docker.
```
Comment (Crell) [inline src/frameworks/ez.md]: As above, remove "just"

Comment (Crell) [reply src/frameworks/ez.md]: Also Docker is capitalized.

File: src/frameworks/ez.md [hunk 8]
```diff
@@ -1,21 +1,146 @@
...
+
+```bash
+~/ez init
+```
+
+or just create the docker stack based on an existing project
```
Comment (Crell) [inline src/frameworks/ez.md]: "just"

Comment (Crell) [reply src/frameworks/ez.md]: And "Docker"

File: src/frameworks/ez.md [hunk 9]
```diff
@@ -1,21 +1,146 @@
...
- * The [platformsh.php](https://github.com/ezsystems/ezplatform/blob/master/app/config/env/platformsh.php) configuration file, which does the work of mapping Platform.sh environment variables into eZ Platform.  It also will automatically enable Redis-based cache and session support if detected.
+ * The [platformsh.php](https://github.com/ezsystems/ezplatform/blob/master/app/config/env/platformsh.php) configuration
+  file, which does the work of mapping Platform.sh environment variables into eZ Platform.  It also will automatically 
+  enable Redis-based cache and session support if detected.
+
+
+## Local Development with eZ Platform 2.x and later
+
+eZ systems provide a tool called [eZ Launchpad](https://ezsystems.github.io/launchpad/) for local development that
+improves Developer eXperience and reduces the container management. eZ Launchpad is ready to work with Platform.sh.
```
Comment (Crell) [inline src/frameworks/ez.md]: This paragraph should discuss how it's build atop Docker for local dev, as otherwise Docker just sort of appears in the instructions further down with no explanation.

File: src/frameworks/ez.md [hunk 10]
```diff
@@ -1,21 +1,146 @@
...
+Then you can just start to use it to initialize your eZ Platform project on top docker.
+
+```bash
+~/ez init
+```
+
+or just create the docker stack based on an existing project
+
+```bash
+git clone yourproject.git application; cd application; ~/ez create
```
Comment (Crell) [inline src/frameworks/ez.md]: Break into multiple lines, please, for clarity.

File: src/frameworks/ez.md [hunk 11]
```diff
@@ -1,21 +1,146 @@
...
+```
+
+You will find more details on the [eZ Launchpad documentation](https://ezsystems.github.io/launchpad/).
+
+At this time you will have a working eZ Platform application with many services including Varnish, Solr, Redis etc.
+
+### Platform.sh integration
+
+To generate the key files for Platform.sh (`.platform.app.yaml` and `.platform`) just run:
```
Comment (Crell) [inline src/frameworks/ez.md]: "just"

File: src/frameworks/ez.md [hunk 12]
```diff
@@ -1,21 +1,146 @@
...
+
+```bash
+~/ez platformsh:setup
+```
+
+eZ Launchpad will generate the files for you and you are then totally free to finetune them.
```
Comment (Crell) [inline src/frameworks/ez.md]: "fine tune", two words.

File: src/frameworks/ez.md [hunk 13]
```diff
@@ -1,21 +1,146 @@
...
+
+> Note that SolR is not currently managed. You will have to set it up manually for Platform.sh.
```
Comment (Crell) [inline src/frameworks/ez.md]: "Solr", not "SolR".

Also, what does this mean?  Set it up manually how?  Does this mean I need to use a tunnel, or install a local copy, or...?  This is very unclear.

File: src/frameworks/ez.md [hunk 14]
```diff
@@ -1,21 +1,146 @@
...
+
+#### Environment variables (optional)
+
+One advantage among others with eZ Launchpad(thanks to Docker) and Platform.sh is you can easily reproduce the 
+Platform.sh environment variables in the `provisioning/dev/docker-compose.yml` file. It allows you to test locally
+the Platform.sh context so you stay consistent everywhere.
```
Comment (Crell) [inline src/frameworks/ez.md]: This sentence feels very clumsy.  Try:

"eZ Launchpad allows you to define environment variables in the `provisioning/dev/docker-compose.yml` file.  You may use that to set Platform.sh variables [link that to the variables page] to match Platform.sh environments so that you can keep your environment behavior in sync."

An example here would also be good.

File: src/frameworks/ez.md (hunk 1 above)
Comment (Plopix) [reply src/frameworks/ez.md]: done

File: src/frameworks/ez.md (hunk 2 above)
Comment (Plopix) [reply src/frameworks/ez.md]: done

File: src/frameworks/ez.md (hunk 3 above)
Comment (Plopix) [reply src/frameworks/ez.md]: done

File: src/frameworks/ez.md (hunk 4 above)
Comment (Plopix) [reply src/frameworks/ez.md]: right, fixed

File: src/frameworks/ez.md (hunk 5 above)
Comment (Plopix) [reply src/frameworks/ez.md]: fixed

File: src/frameworks/ez.md (hunk 6 above)
Comment (Plopix) [reply src/frameworks/ez.md]: ok clear. done.

File: src/frameworks/ez.md (hunk 7 above)
Comment (Plopix) [reply src/frameworks/ez.md]: done

File: src/frameworks/ez.md (hunk 8 above)
Comment (Plopix) [reply src/frameworks/ez.md]: done as well

File: src/frameworks/ez.md (hunk 9 above)
Comment (Plopix) [reply src/frameworks/ez.md]: I have changed it, let me know.

File: src/frameworks/ez.md (hunk 10 above)
Comment (Plopix) [reply src/frameworks/ez.md]: done

File: src/frameworks/ez.md (hunk 11 above)
Comment (Plopix) [reply src/frameworks/ez.md]: done

File: src/frameworks/ez.md (hunk 12 above)
Comment (Plopix) [reply src/frameworks/ez.md]: fixed

File: src/frameworks/ez.md (hunk 13 above)
Comment (Plopix) [reply src/frameworks/ez.md]: I have put more stuff here to precise. Solr is working, the auto-configuration is not. Let me know what you think of my change.

File: src/frameworks/ez.md (hunk 14 above)
Comment (Plopix) [reply src/frameworks/ez.md]: done it.

File: src/frameworks/ez.md [hunk 15]
```diff
@@ -2,7 +2,7 @@
 
 eZ Platform is a Composer-based PHP CMS, and as such fits well with the Platform.sh model.  As a Symfony-based application its setup is very similar to Symfony.
 
-eZ Platform comes pre-configured for use with Platform.sh in version 1.13 and later and version 2.0.1 and later.  Those are the only versions that are supported.  Appropriate Platform.sh configuration files are included in the eZ Platform application itself, but of course may be modified to suit your particular site if needed.
+eZ Platform comes pre-configured for use with Platform.sh in version 1.13 and later. Version is 2.5 and later is recommended. Those are the only versions that are supported.  Appropriate Platform.sh configuration files are included in the eZ Platform application itself, but of course may be modified to suit your particular site if needed.
```
Comment (Crell) [inline src/frameworks/ez.md]: "Versions 2.5 or later are recommended".  ("Version is" doesn't parse.)

File: src/frameworks/ez.md [hunk 16]
```diff
@@ -18,4 +18,129 @@ In particular, see:
  
  * The [.platform.app.yaml](https://github.com/ezsystems/ezplatform/blob/master/.platform.app.yaml) file, which automatically builds eZ Platform in dev mode or production mode depending on your defined project-level variables.
  * The [.platform](https://github.com/ezsystems/ezplatform/tree/master/.platform) directory
- * The [platformsh.php](https://github.com/ezsystems/ezplatform/blob/master/app/config/env/platformsh.php) configuration file, which does the work of mapping Platform.sh environment variables into eZ Platform.  It also will automatically enable Redis-based cache and session support if detected.
+ * The [platformsh.php](https://github.com/ezsystems/ezplatform/blob/master/app/config/env/platformsh.php) configuration file, which does the work of mapping Platform.sh environment variables into eZ Platform.  It also will automatically  enable Redis-based cache and session support if detected.
```
Comment (Crell) [inline src/frameworks/ez.md]: Not a merge blocker here, but you may want to refactor that file to use the new config-reader library. :smile:

File: src/frameworks/ez.md [hunk 17]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+
+```bash
+~/ez platformsh:setup
+```
+
+eZ Launchpad will generate the files for you and you are then totally free to fine tune them.
+
+#### Solr specificity
+
+Solr is fully functional with eZ Launchpad but its auto-configuration for Platform.sh is not currently managed. You will have to set it up manually following the current documenation here: https://github.com/ezsystems/ezplatform/blob/master/.platform/services.yaml#L37.
```
Comment (Crell) [inline src/frameworks/ez.md]: OK, I follow now.  I think "but it is not enabled by default" would be the cleaner English here.  "Auto-configuration" implies some sort of Solr discovery black magic, which is not the case here.

File: src/frameworks/ez.md [hunk 18]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+
+Actions needed are:
+
+ * generare the Solr configuration thanks to the script provided by eZ Systems
```
Comment (Crell) [inline src/frameworks/ez.md]: "generate", not "generare". :smile:

File: src/frameworks/ez.md [hunk 19]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+ * put the result in the `.platform` at the root of your project
+ * add the service in the `.platform/services.yaml`
+ * add the relationship in the `.platform.app..yaml`
```
Comment (Crell) [inline src/frameworks/ez.md]: This list should not be indented.  The bullet points are sufficient.

Also, since each line is a complete sentence it should be capitalized and end with a period.

File: src/frameworks/ez.md [hunk 20]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+# provisioning/dev/docker-compose.yml
+engine:
+    environment:
+        - ASIMPLEVARIABLE=avalue
+        - PLATFORM_RELATIONSHIPS=A_BASE64_ENCODED_VALUE
+```
+
+### Local development with Platform.sh
+
+Thanks to eZ Launchpad we are able to be work 100% locally: [untethered](/gettingstarted/local/untethered.md). We have the whole project working offline on our local machine.
```
Comment (Crell) [inline src/frameworks/ez.md]: Avoid the use of "we" to refer to the user.  The user is "you".  "We" means Platform.sh, Inc.  (Or whatever the French equivalent of "Inc" is.)

File: src/frameworks/ez.md [hunk 21]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+
+> Platform.sh also provides a smooth SSH tunnels integration described in the 
+[tethered](/gettingstarted/local/tethered.md) page. 
+
+Local services are ensured by the Docker stack but there are minimum day-to-day tasks that you might need with Platform.sh.
+
+Main ones being:
```
Comment (Crell) [inline src/frameworks/ez.md]: "The main ones are:"

File: src/frameworks/ez.md [hunk 22]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+```
+
+### Local development with Platform.sh
+
+Thanks to eZ Launchpad we are able to be work 100% locally: [untethered](/gettingstarted/local/untethered.md). We have the whole project working offline on our local machine.
+
+> Platform.sh also provides a smooth SSH tunnels integration described in the 
+[tethered](/gettingstarted/local/tethered.md) page. 
+
+Local services are ensured by the Docker stack but there are minimum day-to-day tasks that you might need with Platform.sh.
```
Comment (Crell) [inline src/frameworks/ez.md]: "Local services are provided..."

File: src/frameworks/ez.md [hunk 23]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+
+Main ones being:
+
+- **Downstream database synchronization**: Getting it from the remote to the local.
+- **Downstream file storage synchronization**: Getting it from the remote to the local.
```
Comment (Crell) [inline src/frameworks/ez.md]: Use * for bullet points, not -, for consistency.

File: src/frameworks/ez.md [hunk 24]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+
+To help you with that, Platform.sh provides a CLI that you probably already have, if you don’t
```
Comment (Crell) [inline src/frameworks/ez.md]: "have, if" should be a new sentence, ending in a colon.  And then blank line between the text and code sample.

File: src/frameworks/ez.md [hunk 25]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+```bash
+curl -sS https://platform.sh/cli/installer | php
+```
+
+> More information: [Command Line Interface](/gettingstarted/cli.md)
```
Comment (Crell) [inline src/frameworks/ez.md]: Actually I'd just remove the install command entirely and say "If you don't, see the install guide" and link "install guide" to the cli page.

File: src/frameworks/ez.md [hunk 26]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+
+
+Combine together, eZ Launcphad and Platform.sh CLI make those actions straight forward and simple.
```
Comment (Crell) [inline src/frameworks/ez.md]: Combined.

File: src/frameworks/ez.md [hunk 27]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+```bash
+platform db:dump --gzip -f ezplatform.sql.gz -d data/ -y
+platform mount:download -m ezplatform/web/var --target=ezplatform/web/var/ -y
+~/ez/importdata
+```
+
+The two first lines get the remote database and storage from the remote environment and stores it locally in `data/`. The third tells to eZ Launchpad to import those data in the Docker stack.
+
+
+> Regarding the storage(images and files) synchronization: **it is optional**, eZ Platform provides a
```
Comment (Crell) [inline src/frameworks/ez.md]: Space after "storage".

Comment (Crell) [reply src/frameworks/ez.md]: And period after "optional".

Although really, I'd reword the sentence to remove the "Regarding" clause.  "The storage synchronization is optional. eZ Platform provides..."

File: src/frameworks/ez.md [hunk 28]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+The two first lines get the remote database and storage from the remote environment and stores it locally in `data/`. The third tells to eZ Launchpad to import those data in the Docker stack.
+
+
+> Regarding the storage(images and files) synchronization: **it is optional**, eZ Platform provides a 
+[placeholder generator mechanism](https://doc.ezplatform.com/en/latest/guide/images/#setting-placeholder-generator)
+which allows you to forget about the real images for your local.
+
+#### Other automations
+
+eZ Launchpad wraps Docker, then it allows you to stay on the host and execute actions in containers.
```
Comment (Crell) [inline src/frameworks/ez.md]: This sentence doesn't parse.  I think you mean: "eZ Launchpad serves as a wrapper that allows you to run console commands from within the container without logging into it explicitly.  For example:"

File: src/frameworks/ez.md [hunk 29]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+
+> Regarding the storage(images and files) synchronization: **it is optional**, eZ Platform provides a 
+[placeholder generator mechanism](https://doc.ezplatform.com/en/latest/guide/images/#setting-placeholder-generator)
+which allows you to forget about the real images for your local.
+
+#### Other automations
+
+eZ Launchpad wraps Docker, then it allows you to stay on the host and execute actions in containers. 
+
+> Using Docker you have a container for PHP, some this container needs to execute commands not your host.
```
Comment (Crell) [inline src/frameworks/ez.md]: This Note can be removed entirely.

File: src/frameworks/ez.md [hunk 30]
```diff
@@ -18,4 +18,129 @@ In particular, see:
...
+```
+
+The two first lines get the remote database and storage from the remote environment and stores it locally in `data/`. The third tells to eZ Launchpad to import those data in the Docker stack.
+
+
+> Regarding the storage(images and files) synchronization: **it is optional**, eZ Platform provides a 
+[placeholder generator mechanism](https://doc.ezplatform.com/en/latest/guide/images/#setting-placeholder-generator)
+which allows you to forget about the real images for your local.
+
+#### Other automations
```
Comment (Crell) [inline src/frameworks/ez.md]: I question if this section is even needed here; isn't this just part of the Launchpad documentation linked above?  This has nothing to do with Psh.

File: src/frameworks/ez.md (hunk 16 above)
Comment (Plopix) [reply src/frameworks/ez.md]: true, cc @andrerom as it is not really related to this PR ;)

File: src/frameworks/ez.md (hunk 20 above)
Comment (Plopix) [reply src/frameworks/ez.md]: thanks @Crell for all the typos (including this one) you are fixing. I do very well know the difference between `you` and `we`.

File: src/frameworks/ez.md (hunk 30 above)
Comment (Plopix) [reply src/frameworks/ez.md]: I removed it. And moved a line to mention this "wrapper" thing at the very beginning.
Let me know

**Comment by Plopix** [comment]
Thank you @Crell I did the changes!
